pudb/py3compat.py

```diff
@@ -11,6 +11,43 @@
 
     def execfile(fname, globs, locs=None):
         exec(compile(open(fname).read(), fname, "exec"), globs, locs or globs)
+
+    # {{{ container metaclasses
+
+    from abc import ABC
+
+    class PudbCollection(ABC):
+        @classmethod
+        def __subclasshook__(cls, c):
+            if cls is PudbCollection:
+                return all([
+                    any("__contains__" in b.__dict__ for b in c.__mro__),
+                    any("__iter__" in b.__dict__ for b in c.__mro__),
+                ])
+            return NotImplemented
+
+    class PudbSequence(ABC):
+        @classmethod
+        def __subclasshook__(cls, c):
+            if cls is PudbSequence:
+                return all([
+                    any("__getitem__" in b.__dict__ for b in c.__mro__),
+                    any("__iter__" in b.__dict__ for b in c.__mro__),
```

Comment:
Given that sequences end up with numerical indices, I'm worried that these tests aren't really specific enough. (E.g. `set` and `dict` would match.)

Reply:
Yup this is true: the way we're set up right now we're dependent on trying more specific types first.